src/components/TextInput/BaseTextInput.js

```diff
@@ -21,381 +21,380 @@ import isInputAutoFilled from '../../libs/isInputAutoFilled';
+    const [width, setWidth] = useState();
+
+    const input = useRef(null);
+    const isLabelActive = useRef(initialActiveLabel);
+
+    useEffect(() => {
+        let appStateSubscription;
+        if (props.disableKeyboard) {
+            appStateSubscription = AppState.addEventListener('change', dismissKeyboardWhenBackgrounded);
         }
 
         // We are manually managing focus to prevent this issue: https://github.com/Expensify/App/issues/4514
-        if (!this.props.autoFocus || !this.input) {
+        if (!props.autoFocus || !input.current) {
             return;
         }
 
-        if (this.props.shouldDelayFocus) {
-            this.focusTimeout = setTimeout(() => this.input.focus(), CONST.ANIMATED_TRANSITION);
+        let focusTimeout;
+        if (props.shouldDelayFocus) {
+            focusTimeout = setTimeout(() => input.current.focus(), CONST.ANIMATED_TRANSITION);
             return;
         }
-        this.input.focus();
-    }
+        input.current.focus();
 
-    componentDidUpdate(prevProps) {
-        // Activate or deactivate the label when value is changed programmatically from outside
-        const inputValue = _.isUndefined(this.props.value) ? this.input.value : this.props.value;
-        if ((_.isUndefined(inputValue) || this.state.value === inputValue) && _.isEqual(prevProps.selection, this.props.selection)) {
-            return;
-        }
+        return () => {
+            if (focusTimeout) {
+                clearTimeout(focusTimeout);
+            }
 
-        // eslint-disable-next-line react/no-did-update-set-state
-        this.setState({value: inputValue, selection: this.props.selection});
+            if (!props.disableKeyboard || !appStateSubscription) {
+                return;
+            }
 
-        // In some cases, When the value prop is empty, it is not properly updated on the TextInput due to its uncontrolled nature, thus manually clearing the TextInput.
-        if (inputValue === '') {
-            this.input.clear();
-        }
-
-        if (inputValue) {
-            this.activateLabel();
-        } else if (!this.state.isFocused) {
-            this.deactivateLabel();
-        }
-    }
-
-    componentWillUnmount() {
-        if (this.focusTimeout) {
-            clearTimeout(this.focusTimeout);
-        }
-
-        if (!this.props.disableKeyboard || !this.appStateSubscription) {
+            appStateSubscription.remove();
+        };
+    }, [props.autoFocus, props.disableKeyboard, props.shouldDelayFocus]);
+
+    const propIsDisabled = props.disabled;
+    const propOnPress = props.onPress;
+    const onPress = useCallback(
+        (event) => {
+            if (propIsDisabled) {
+                return;
+            }
+
+            if (propOnPress) {
+                propOnPress(event);
+            }
+
+            if (!event.isDefaultPrevented()) {
+                input.current.focus();
+            }
+        },
+        [propIsDisabled, propOnPress],
```

Comment:
if we will not use desctructive props you should remove this items and use direct props.disabled, and props.onPress. in current implementation you will re-create onPress callback on any re-render - which may cause performance and memory issues